### PR TITLE
Update script for integration tests on Databricks [databricks]

### PR DESCRIPTION
**Pull request by NvTimLiu**
To fix issues :  https://github.com/NVIDIA/spark-rapids/issues/7473, https://github.com/NVIDIA/spark-rapids/issues/7773, https://github.com/NVIDIA/spark-rapids/issues/7776, 

1, Create the file cudf_udf_test.sh for the cudf_udf tests. 

2, Install pytest packages via pip.

3, Update init_cudf_udf.sh to support running under both root and non-root permission.

### CODE REVIEW

**Comment by NvTimLiu** [comment]
For https://github.com/NVIDIA/spark-rapids/issues/7473, test can PASS without any conda pre-installation:

```
.  jenkins/databricks/test.sh

......

[gw0] [ 98%] PASSED ../../src/main/python/hash_aggregate_test.py::test_hash_grpby_sum_full_decimal[{'spark.rapids.sql.variableFloatAgg.enabled': 'true', 'spark.rapids.sql.castStringToFloat.enabled': 'true', 'spark.rapids.sql.batchSizeBytes': '250'}-[('a', RepeatSeq(Short)), ('b', Decimal(38,-10)), ('c', Decimal(38,-10))]][IGNORE_ORDER, INCOMPAT, APPROXIMATE_FLOAT] 
[gw3] [100%] PASSED ../../src/main/python/hash_aggregate_test.py::test_hash_grpby_sum_full_decimal[{'spark.rapids.sql.variableFloatAgg.enabled': 'true', 'spark.rapids.sql.castStringToFloat.enabled': 'true', 'spark.rapids.sql.hashAgg.replaceMode': 'partial'}-[('a', RepeatSeq(Short)), ('b', Decimal(38,-10)), ('c', Decimal(38,-10))]][IGNORE_ORDER, INCOMPAT, APPROXIMATE_FLOAT, ALLOW_NON_GPU(HashAggregateExec,AggregateExpression,UnscaledValue,MakeDecimal,AttributeReference,Alias,Sum,Count,Max,Min,Average,Cast,StddevPop,StddevSamp,VariancePop,VarianceSamp,NormalizeNaNAndZero,GreaterThan,Literal,If,EqualTo,First,SortAggregateExec,Coalesce,IsNull,EqualNullSafe,PivotFirst,GetArrayItem,ShuffleExchangeExec,HashPartitioning)] 

generated xml file: /home/ubuntu/spark-rapids/integration_tests/target/run_dir-20230218180001-a9eZ/TEST-pytest-1676743201055115392.xml -
=================== 64 passed, 2 skipped in 99.73s (0:01:39) ===================
```


For https://github.com/NVIDIA/spark-rapids/issues/7773, moved `cudf_udf` tests out into the file ` jenkins/databricks/cudf_udf_test.sh` from ` jenkins/databricks/test.sh`; 



For https://github.com/NVIDIA/spark-rapids/issues/7776, keep to use the default Databricks python in `jenkins/databricks/test.sh` for spark-rapids integration tests, only apply workaround of cudf_udf conda installing with `python=3.8`  for the file ` jenkins/databricks/cudf_udf_test.sh`

File: jenkins/databricks/setup.sh [hunk 1]
```diff
@@ -36,32 +36,17 @@ if [ -f $SPARK_HOME/conf/spark-env.sh ]; then
...
-        numpy
-        pyarrow
-    )
-
-    echo creating symlinks to override conflicting packages
-    for p in "${TO_PATCH[@]}"; do
-        ln -f -s ${CONDA_SITE_PATH}/${p} ${PATCH_PACKAGES_PATH}
-    done
-fi
+python_versionn=$(${PYSPARK_PYTHON} -c 'import sys; print("python{}.{}".format(sys.version_info.major, sys.version_info.minor))')
```
Comment (pxLi) [inline jenkins/databricks/setup.sh:48]: nit: remove trailing `n`

Comment (NvTimLiu) [reply jenkins/databricks/setup.sh]: good catch!

**Comment by NvTimLiu** [comment]
@jlowe  @tgravescs  @revans2  Can you please help to review? Thanks!

**Review by tgravescs — APPROVED** [review]
I'm assuming the test.sh isn't called when we run integration tests.. ie setup.sh has same pip install, so that is only used with run_it.sh that doesn't use test.sh?

**Comment by NvTimLiu** [comment]
> 

Correct, setup.sh is only used in the run_it.sh

test.sh will not call run_it.sh, 

run_it.sh is another IT scripts, similar to test.sh,

**Comment by NvTimLiu** [comment]
build

**Pull request merged**